docarray/array/doc_list/sequence_indexing_mixin.py

```diff
@@ -11,8 +11,11 @@
     cast,
     no_type_check,
     overload,
+    List,
 )
 
+from typing_extensions import SupportsIndex
```

Comment:
what is this SupportsIndex ?

Reply:
I think it's a fancy alternative for int